jbmc/src/jbmc/jbmc_parse_options.cpp

```diff
@@ -529,6 +529,9 @@ int jbmc_parse_optionst::doit()
       *this, options, get_message_handler());
     lazy_goto_model.initialize(cmdline, options);
 
+    class_hierarchy = util_make_unique<class_hierarchyt>();
+    (*class_hierarchy)(lazy_goto_model.symbol_table);
```

Comment:
:confused: Can't think of a better way to write this, but is a bit of a head scratcher. :bulb: Perhaps pull into a static method in `class_hierachy.h`: std::unique_ptr<class_hierachyt> create_hierachy(const symbol_tablet &)`?

Reply:
Added an extra constructor, take another look?